ArmRobot/Assets/Scripts/DomainRandomization.cs

```diff
@@ -0,0 +1,218 @@
+        RobotController robotController = robot.GetComponent<RobotController>();
+        float rangeRotation = -rotationAngle + 2 * rotationAngle * Random.value;
+        float[] rotation = {rangeRotation, rangeRotation, rangeRotation, rangeRotation, rangeRotation, rangeRotation};
+        robotController.ForceJointsToRotations(rotation);
+    }
+
+    List<GameObject> CreateRandomListCylinders()
+    {
+        // this function is designed to create a list of a random number of cylinders and instantiate them 
+        float minimumDistanceBetweenObjects = scaleObject + minimumDistanceWithoutObjects;
+        int randomNumberCylinders = 1 + Random.Range(1, nbMaxCylinders); 
+        List<GameObject> listOfCylinders = new List<GameObject>();
+        for (var i = 0; i < randomNumberCylinders; i++) {
+            GameObject cylinder  = GameObject.CreatePrimitive(PrimitiveType.Cylinder);
+            
+            cylinder.transform.position = new Vector3(robotMaxReach - 2*minimumDistanceWithoutObjects , 
+                                                yAltitudeTable, robotMaxReach - 2*minimumDistanceWithoutObjects - minimumDistanceBetweenObjects*i);
+            
+            
+            cylinder.transform.localScale = new Vector3(scaleObject, scaleObject, scaleObject);
+            cylinder.AddComponent<ColorRandomizer>();
+            cylinder.AddComponent<RandomizerPositionObject>();
+            cylinder.tag = "Cylinder";
+            ColorRandomizer colorRandomizer = cylinder.GetComponent<ColorRandomizer>();
+            colorRandomizer.ChangeColor();
+            listOfCylinders.Add(cylinder);
+        }
+        return listOfCylinders;
+    }
+
+    List<GameObject> CreateRandomListSpheres()
+    {
+        // this function is designed to create a list of a random number of spheres and instantiate them 
+        float minimumDistanceBetweenObjects = scaleObject + minimumDistanceWithoutObjects;
+        int randomNumberSpheres = 1 + Random.Range(1, nbMaxSpheres); 
+        List<GameObject> listOfSpheres = new List<GameObject>();
+        for (var i = 0; i < randomNumberSpheres; i++) {
+            GameObject sphere  = GameObject.CreatePrimitive(PrimitiveType.Sphere);
+
+            sphere.transform.position = new Vector3(robotMaxReach - 2*minimumDistanceWithoutObjects - minimumDistanceBetweenObjects, 
+                                                yAltitudeTable, robotMaxReach - 2*minimumDistanceWithoutObjects - minimumDistanceBetweenObjects*i);
+            
+            sphere.transform.localScale = new Vector3(scaleObject, scaleObject, scaleObject);
+            sphere.AddComponent<ColorRandomizer>();
+            sphere.AddComponent<RandomizerPositionObject>();
+            sphere.tag = "Sphere";
+            ColorRandomizer colorRandomizer = sphere.GetComponent<ColorRandomizer>();
+            colorRandomizer.ChangeColor();
+            listOfSpheres.Add(sphere);
```

Comment:
It would be even more readable and modular to break out another method to create one cylinder, like GameObject CreateSphere(Vector3 position, Vector3 scale) that we could call from this method.